### PR TITLE
Add v2 SDK and migrate to viem

### PR DESCRIPTION
**Pull request by ccummings**
- Migrates v1 to viem
- Adds a v2 SDK

### CODE REVIEW

**Comment by ryespresso** [comment]
@ccummings wow, thanks for this. will review asap

File: src/v1.ts [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+export const DelegateV1 = class {
+  publicClient: PublicClient;
+  walletClient: WalletClient;
+  account?: Account | Address;
+  contractConfig = {
+    address: ADDRESS,
+    abi: ABI,
+  };
+
+  constructor(chain: Chain = mainnet, userTransport: Transport = http(), account?: Account | Address) {
```
Comment (ryespresso) [inline src/v1.ts]: Would be nice to make `chain` optional and change the order to:

```
constructor(
    userTransport: Transport = http(), 
    chain: Chain = mainnet, 
    account?: Account | Address
)
```

File: src/v2.ts [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+export const DelegateV2 = class {
+  publicClient: PublicClient;
+  walletClient: WalletClient;
+  account?: Account | Address;
+  contractConfig = {
+    address: ADDRESS,
+    abi: ABI,
+  };
+
+  constructor(chain: Chain = mainnet, userTransport: Transport = http(), account?: Account | Address) {
```
Comment (ryespresso) [inline src/v2.ts]: Would be nice to make `chain` optional and change the order to:

```
constructor(
    userTransport: Transport = http(), 
    chain: Chain = mainnet, 
    account?: Account | Address
)
```

**Review by ryespresso — CHANGES_REQUESTED** [review]
Absolutely killed it. Thank you for the pull. Just one request to make it easier on people. Interested to hear what you think.

File: src/v1.ts (hunk 1 above)
Comment (ccummings) [reply src/v1.ts]: I like that, good call.

**Review by ryespresso — APPROVED** [review]

**Pull request merged**